### DISCUSSION TITLE
Flask sessions do not work in deta

### DISCUSSION THREAD

**Discussion by Pranav435** [deta/cloud-docs#615] [Help]
Hi There,

I have a flask app running in data.
I have to store something via sessions. It works when running the app locally. However, the sessions do not persist when deploying to data.

Any help would be appreciated.

**Comment by MythicalJay** [comment]
Similar thing for me too. I am trying to have a session as well as store an external API KEY (for stock data). Is there an option to store other API keys as external variables?

**Comment by jnsougata** [comment]
Sessions won't work on micros as they are serverless. One thing you can do is store the states inside base with an expiring key.

**Comment by MystPi** [comment]
`cookie-session` (in Express) stores data on the client inside of a cookie, yet it doesn't work on Deta, only locally.

**Reply by MystPi** [reply]
It seems like sometimes the session gets stored on the client, but most of the time it doesn't! It's very frustrating.
